http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/VPCHelper.scala

```diff
@@ -16,40 +16,43 @@ class VPCHelper(config: VPCHelperConfig,
                 googleProjectDAO: GoogleProjectDAO,
                 googleComputeService: GoogleComputeService[IO])(implicit contextShift: ContextShift[IO]) {
 
-  private[util] def getVPCSettingsFromProjectLabel(googleProject: GoogleProject): IO[Option[VPCConfig]] =
-    IO.fromFuture(IO(googleProjectDAO.getLabels(googleProject.value))).map { labelMap =>
-      labelMap.get(config.projectVPCSubnetLabelName).map(VPCSubnet) orElse
-        labelMap.get(config.projectVPCNetworkLabelName).map(VPCNetwork)
-    }
-
-  // TODO move toward creating our own dedicated subnet instead of using the default
-  private[util] def createVPCSubnet(googleProject: GoogleProject): IO[VPCConfig] = IO.pure(VPCConfig.default)
-
   def getOrCreateVPCSettings(googleProject: GoogleProject): IO[VPCConfig] =
     for {
-      fromProjectLabels <- getVPCSettingsFromProjectLabel(googleProject)
-      res <- fromProjectLabels.fold(createVPCSubnet(googleProject))(IO.pure)
+      // In priority order, use:
+      // 1. the subnet specified by project label
+      // 2. the network specified by project label
+      // 3. the default network
+      // TODO in the future create a dedicated subnet instead of using the default
+      projectLabels <- IO.fromFuture(IO(googleProjectDAO.getLabels(googleProject.value)))
+      subnetFromLabel = projectLabels.get(config.projectVPCSubnetLabelName).map(VPCSubnet)
+      networkFromLabel = projectLabels.get(config.projectVPCNetworkLabelName).map(VPCNetwork)
+      res = subnetFromLabel.orElse(networkFromLabel).getOrElse(VPCConfig.defaultNetwork)
     } yield res
 
-  def getOrCreateFirewallRule(googleProject: GoogleProject, regionName: RegionName, vpcConfig: VPCConfig)(
+  def getOrCreateFirewallRule(googleProject: GoogleProject)(
     implicit ev: ApplicativeAsk[IO, TraceId]
   ): IO[Unit] =
     for {
-      rule <- googleComputeService.getFirewallRule(googleProject, config.firewallRuleName)
-      _ <- rule.fold(
-        googleComputeService.addFirewallRule(googleProject, buildFirewall(googleProject, regionName, vpcConfig))
-      )(
-        _ => IO.unit
-      )
+      ruleOpt <- googleComputeService.getFirewallRule(googleProject, config.firewallRuleName)
+      _ <- if (ruleOpt.isEmpty) {
+        // Firewall rules don't work with a subnet, so we have to pass a network. This is
```

Comment:
The issue in the last PR is that firewall rules require a network, not a subnet… sigh

Reply:
so what was the initial bug?

Reply:
I'm a bit confused how it used to work before we used `VPCHelper`/`GoogleComputeService`. Looking at old code, it seemed we passed the subnet to the firewall creation request in `HttpGoogleComputeDAO` (which is why I initially did that in my last PR). It's possible that the old code wasn't correct either and it was working by accident somehow.

Reply:
This is so tricky 😞

Reply:
It really is.. before the post-mortem I'll try and better understand this change in behavior.